### ISSUE TITLE
mo.ui.slider with very small step value does not work

### ISSUE THREAD

**Issue by vangberg** [marimo-team/marimo#2593]
### Describe the bug

If I create a slider with a step value of `10**-7` the slider does not respond. A step value of `10**-06` works fine.

https://github.com/user-attachments/assets/177b352c-8689-46f0-a926-043cda5133d1

### Environment

{
  "marimo": "0.9.6",
  "OS": "Darwin",
  "OS Version": "22.3.0",
  "Processor": "arm",
  "Python Version": "3.12.6",
  "Binaries": {
    "Browser": "129.0.6668.91",
    "Node": "v22.9.0"
  },
  "Dependencies": {
    "click": "8.1.7",
    "importlib-resources": "missing",
    "jedi": "0.19.1",
    "markdown": "3.7",
    "pygments": "2.18.0",
    "pymdown-extensions": "10.10.1",
    "ruff": "0.6.7",
    "starlette": "0.39.0",
    "tomlkit": "0.13.2",
    "typing-extensions": "4.12.2",
    "uvicorn": "0.30.6",
    "websockets": "12.0"
  },
  "Optional Dependencies": {}
}

### Code to reproduce

import marimo as mo

works = mo.ui.slider(10**-7, 10**-5, 10**-6, show_value=True)

fails = mo.ui.slider(10**-7, 10**-5, 10**-7, show_value=True)

**Comment by mscolnick** [comment]
This is coming from an upstream library and looking through their code, its likely a rounding issue.

The seemed to fix this for small numbers, but I am guessing not this small.
https://github.com/radix-ui/primitives/issues/462
https://github.com/radix-ui/primitives/pull/463

**Comment by mscolnick** [comment]
we may have better luck switching to react-aria-components 

https://react-spectrum.adobe.com/react-aria/Slider.html

**Comment by rubiagatra** [comment]
I am using 

```
dependencies = [
    "marimo>=0.14.12",
]
```

still cannot use the slider, even though the PR has already merged. Would you like me to explore the React ARIA slide? @mscolnick ?

<img width="769" height="321" alt="Image" src="https://github.com/user-attachments/assets/d8511e0f-f841-4cd4-8f01-31b7cfbf3db2" />

**Comment by mscolnick** [comment]
@rubiagatra, yes if you are open to exploring migrating ot react-aria-components, that would be really appreciated. 

it would be good first to check if they support this small of a step, before doing doing the full migration

**Comment by mscolnick** [comment]
Use `steps=` instead of `step=`. Passing an explicit list runs the slider in index space, sidestepping the radix rounding error:

```py
mo.ui.slider(steps=np.arange(1e-7, 1e-5 + 1e-7, 1e-7).tolist(), show_value=True)
```

**Comment by bhavya632** [comment]
Hi @mscolnick — I'd be interested in picking this up, but wanted to check the scope first.

Given your recent note about using steps= with an explicit list as a workaround, is the full radix → react-aria-components migration still the direction you'd want here? Or is the steps= approach now the preferred answer and this is closer to resolved?

I also saw there was some earlier interest in taking on the react-aria migration — is that still active, or is this open? Happy to take it on if it's still wanted; just want to confirm the scope is real before starting, since swapping the slider library is a fair bit more than a one-line fix.
